test/replication-luatest/election_fencing_test.lua

```diff
@@ -289,3 +282,73 @@ g_sync.test_fencing = function(g)
     luatest.assert_equals(ret, {},
         'Sync write rollbacked after new leader discovered')
 end
+
+g_mode.before_all(function(g)
+    start(g)
+    box_cfg_update({g.server_3}, {replication = {}})
+    wait_disconnected(g.server_1, g.server_3)
+    wait_disconnected(g.server_2, g.server_3)
+end)
+
+g_mode.after_all(stop)
+
+g_mode.test_fencing_mode = function(g)
+    local timeout = 0.5
+    box_cfg_update({g.server_1, g.server_2}, {
+        election_fencing_mode = g.params.election_fencing_mode,
+        replication_timeout = timeout,
+    })
+
+    g.server_1:exec(function(input, output)
+        rawset(_G, 'proxy', require('test.luatest_helpers.proxy.proxy'):new({
+            input_path = input,
+            output_path = output,
+        }))
+        _G.proxy:start({force = true})
+    end, {
+        server.build_instance_uri('server_1_proxy'),
+        server.build_instance_uri('server_1')})
+
+    local proxied_replication = {
+        server.build_instance_uri('server_1_proxy'),
+        server.build_instance_uri('server_2'),
+    }
+
+    box_cfg_update({g.server_2}, {replication = {}})
+    wait_disconnected(g.server_1, g.server_2)
+    box_cfg_update({g.server_2}, {replication = proxied_replication})
+    wait_connected(g.server_1, g.server_2)
+
+    promote(g.server_1)
+    local leader_id = g.server_1:instance_id()
+    wait_sync(g.server_1, {g.server_2})
+
+    box_cfg_update({g.server_2}, {
+        election_mode = 'candidate',
+        replication_synchro_quorum = 1,
+    })
+
+    g.server_1:exec(function()
+        _G.proxy:pause()
+    end)
+
+    g.server_1:wait_election_state('follower')
+
+    -- Give folower some time to notice leader disconnection.
+    require('fiber').sleep(timeout / 10)
```

Comment:
Well, there is no guarantee. If we don't wait at all, it makes _soft fencing_ test flaky, because follower might be to slow to start the elections and declare itself leader, after old leader steps down. If it is closer to replication_timeout it will make _strict fencing_ test flaky, because it might be enough even with strong fencing for follower to be elected after old leader steps down. So, we need a timeout in range (0, replication_timeout). In addition, the shorter we wait, the faster the test. I've chosen replication_timeout/10, because it was short and stable enough.

Reply:
Well, sounds like it will become flaky 100% eventually. Ok, then lets fix it when that happens.